### ISSUE TITLE
Relation manager obeys they $fillable/$guarded, but regular page does not.

### ISSUE THREAD

**Issue by that0n3guy** [octobercms/october#1042]
This bug was driving me crazy, but here is how to duplicate it:

* Add a relationship manager to an "update" form with something (like blogpost model) like `<?= $this->relationRender('comments') ?>`
* Then set a field in fields.yaml in comments like `from_email`.  A related column should not exist in the DB.
* in your comments model, make sure you have `from` field to be jsonable
* make sure you have a `from` column in your database.
* add the accessor and mutators for the `from_email` attribute in your comments model, like so:

```
    public function getFromEmailAttribute() {
        if( isset ($this->from['from_email']) ){
            return $this->from['from_email'];
        }
        return '';
    }

    public function setFromEmailAttribute($value) {
        
        $this->from = array_replace_recursive((array)$this->from, ['from_email' => $value]);
    }
```

* Try to edit a comment from the relation manager on blogpost update form.  It won't complain (no error) and wont save the `from_email` field.

*I've verified that setFromEmailAttribute() is NOT fired via a breakpoint with xdebug.  `getFromEmailAttribute` is working fine though.*

Also, this DOES work if you are creating/editing a comment normally (like on a page like: backend/oca/mypluginname/comments/update/2).  It works perfectly from a normal page... **just not from the relation manager.**

**Comment by that0n3guy** [comment]
Ok, I figured this out, but its still a bug of some kind.

When using the relation manager, it obeys they $fillable/$guarded vars.  So, setFromEmailAttribute is not called if I don't have `from_email` in fillable (if $guarded='*').   BUT the normal page (example: backend/oca/mypluginname/comments/update/2) does not obey this and setFromEmailAttribute is still run even if `from_email` is NOT in fillable.

So, the bug is they don't act the same, relation manager and normal backend page..

**Comment by daftspunk** [comment]
Yes, this is inconsistent and should be fixed.

**Comment by daftspunk** [comment]
Fixed in af657e9399e3ac74ed70a0b35089b24e07cf56e9
